lib/PersonalityProvider.jsm

```diff
@@ -38,6 +56,46 @@ this.PersonalityProvider = class PersonalityProvider {
     this.scores = scores || {};
     this.interestConfig = this.scores.interestConfig;
     this.interestVector = this.scores.interestVector;
+    this.syncAttachment("personality-provider-recipe-attachment");
+    this.syncAttachment("personality-provider-models-attachment");
+  }
+
+  syncAttachment(collection) {
+    RemoteSettings(collection).on("sync", async event => {
+      const {
+        data: {created, updated, deleted},
+      } = event;
+
+      // Remove every removed attachment.
+      const toRemove = deleted.concat(updated.map(u => u.old));
+      await Promise.all(
+        toRemove.map(record => {
+          const {attachment: {filename}} = record;
+
+          const path = OS.Path.join(OS.Constants.Path.localProfileDir, filename);
+          return OS.File.remove(path, {ignoreAbsent: true});
+        })
+      );
+
+      // Download every new/updated attachment.
+      const toDownload = created.concat(updated.map(u => u.new));
+      await Promise.all(
+        toDownload.map(async record => {
+          const {attachment: {location, filename}} = record;
+
+          // This should be using this line:
+          // const resp = await fetch(`${baseAttachmentsURL}${location}`);
+          // but see above issue in baseAttachmentsURL
+          // for now I'm hard conding in dev url to test
+          const resp = await fetch(`https://kinto.dev.mozaws.net/attachments/${location}`);
```

Comment:
Maybe it's already the case by default, but we should have `Accept-Encoding: gzip` here to make sure we leverage Cloudfront gzipping

Reply:
OK, I think I figured this out.